File: Searchin_Algorithms/Linear_search.cpp
#include <iostream>
using namespace std;

int LinearSearch(int arr[], int n, int target) {
    for(int i = 0; i < n; i++) {
        if(arr[i] == target) {
            return i;
        }
    }
    return -1;
}

int main() {
    int arr[] = {10, 2, 4, 3, 8, 5, 9, 7};
    int n = sizeof(arr)/sizeof(arr[0]);
    int res = LinearSearch(arr, n, 2);
    if(res == -1) {
        cout << "Target not found";
    }
    else {
        cout << "Target found at index " << res;
    }

    return 0;
}

File: Searchin_Algorithms/Linear_search.py
def linear_search(arr, target):
    for i in range(0, len(arr)):
        if arr[i] == target:
            return i
    return None

def verify(index):
    if index is not None:
        print("Target found at index: ", index)
    else:
        print("Target not found")

arr = [10, 7, 9, 8, 1, 4, 2, 3]

res = linear_search(arr, 12)
verify(res)